src/Map/WeatherTileResourceProvider_P.cpp

```diff
@@ -376,42 +396,221 @@ void OsmAnd::WeatherTileResourceProvider_P::downloadGeoTilesAsync(
 
 std::shared_ptr<OsmAnd::TileSqliteDatabase> OsmAnd::WeatherTileResourceProvider_P::getGeoTilesDatabase()
 {
-    QReadLocker scopedLocker(&_geoDbLock);
+    {
+        QReadLocker geoScopedLocker(&_geoDbLock);
+
+        if (_geoTilesDb)
+            return _geoTilesDb;
+    }
+    {
+        QWriteLocker geoScopedLocker(&_geoDbLock);
 
-    return _geoTilesDb;
+        _geoTilesDb = createGeoTilesDatabase();
+        return _geoTilesDb;
+    }
 }
 
 std::shared_ptr<OsmAnd::TileSqliteDatabase> OsmAnd::WeatherTileResourceProvider_P::getRasterTilesDatabase(BandIndex band)
 {
+    QString key = dateTime.toString(QStringLiteral("yyyyMMdd_hh00"))
```

Comment:
Provider is datatime based, so it cannot have any other datetime. Looks like needs to be reverted.